vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/StaffSearchManagerTest.java

```diff
@@ -0,0 +1,271 @@
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+public class StaffSearchManagerTest {
+
+    @Mock
+    private ISpaceManager spaceManager;
+
+    @Mock
+    private IModuleManager moduleManager;
+
+    @Mock
+    private ISlideManager slideManager;
+
+    @Mock
+    private TextContentBlockRepository textContentBlockRepo;
+
+    @InjectMocks
+    private StaffSearchManager serviceToTest;
+
+    private List<Space> spaces;
+
+    private List<Module> modules;
+
+    private List<Slide> slides;
+
+    private List<Slide> slideTexts;
+
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+        
+        Space space1 = new Space();
+        space1.setId("SPACE_ID_1");
+        Space space2 = new Space();
+        space2.setId("SPACE_ID_2");
+        spaces = new ArrayList<>();
+        spaces.add(space1);
+        spaces.add(space2);
+        
+        Module module1 = new Module();
+        module1.setId("MODULE_ID_1");
+        Module module2 = new Module();
+        module2.setId("MODULE_ID_2");
+        modules = new ArrayList<>();
+        modules.add(module1);
+        modules.add(module2);
+        
+        Slide slide1 = new Slide();
+        slide1.setId("SLIDE_ID_1");
+        Slide slide2 = new Slide();
+        slide2.setId("SLIDE_ID_2");
+        slides = new ArrayList<>();
+        slides.add(slide1);
+        slides.add(slide2);
+        
+        Slide slideText1 = new Slide();
+        slideText1.setId("SLIDETEXT_ID_1");
+        Slide slideText2 = new Slide();
+        slideText2.setId("SLIDETEXT_ID_2");
+        slideTexts = new ArrayList<>();
+        slideTexts.add(slideText1);
+        slideTexts.add(slideText2);
+        
+        ReflectionTestUtils.setField(serviceToTest, "pageSize", 10);
+    }
+
+    @Test
+    public void test_searchInSpaces_success() {
+        
+        Pageable requestedPage = PageRequest.of(0, 10);
+        String search = "space";
+        when(spaceManager.findByNameOrDescription(requestedPage, search)).thenReturn(new PageImpl<Space>(spaces));
+        Page<Space> tempResult = serviceToTest.searchInSpaces(search, 1);
+        assertEquals(2, tempResult.getContent().size());
+        List<String> idList = tempResult.stream().map(space -> space.getId()).collect(Collectors.toList());
+        assertEquals("SPACE_ID_1", idList.get(0));
+        assertEquals("SPACE_ID_2", idList.get(1));
+        verify(spaceManager).findByNameOrDescription(requestedPage, search);
+    }
+
+    @Test
+    public void test_searchInSpaces_pageGreaterThanTotalPages() {
+        
+        Pageable requestedPage = PageRequest.of(6, 10);
+        String search = "space";
+        when(spaceManager.findByNameOrDescription(requestedPage, search)).thenReturn(new PageImpl<Space>(spaces));
+        Pageable requestedPage1 = PageRequest.of(0, 10);
```

Comment:
that doesn't seem to make sense. If there are only 6 pages but nr 7 is requested, the second request should be for page 5, shouldn't?